sklearn/compose/_column_transformer.py

```diff
@@ -417,6 +421,22 @@ def _validate_output(self, result):
                     "The output of the '{0}' transformer should be 2D (scipy "
                     "matrix, array, or pandas DataFrame).".format(name))
 
+    def _index_output(self, Xs):
+        """
+        Record which transformer produced which column.
+        """
+        idx_ = 0
+        self.transformers_output_ = {}
+        for idx, (name, _, _, _) in enumerate(
+            self._iter(fitted=True, replace_strings=True)
+        ):
+            n_columns = Xs[idx].shape[1]
+            self.transformers_output_[name] = list(
+                range(idx_, idx_ + n_columns)
```

Comment:
The order of the names may not have this correspondence. For example:

```py
import numpy as np
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import Normalizer
ct = ColumnTransformer(
    [("norm1", Normalizer(norm='l1'), [1, 2]),
     ("norm2", Normalizer(norm='l1'), [0, 3])])
X = np.array([[0., 1., 2., 2.],
              [1., 1., 0., 1.]])
ct.fit_transform(X)
```

In this case the first name corresponds to the middle two columns.

Reply:
We are not concerned with the order of inputs, but with the order of outputs. Although the first transformer takes in the middle columns of the input data, it produces the first two columns of the output data.

Reply:
Ah yes. This makes sense. 

I would update the docstring for `transformers_output_` to highlight this point.